src/test/Voucher.t.sol

```diff
@@ -0,0 +1,64 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.4;
+
+import "./TestBase.sol";
+
+contract TestVoucher is TestBase {
+
+    function testVoucher(
+        uint64 time,
+        uint128 smallAmount,
+        uint128 largeAmount
+    ) public {
+        vm.assume(smallAmount >= MIN_AMOUNT_FOR_NEW_CLONE);
+        vm.assume(smallAmount < 2**105);
+        vm.assume(largeAmount < 2**105);
+
+        uint256 nftId = nft.mint();
+
+        currency.mint(eoa1, smallAmount);
+
+        vm.startPrank(eoa1);
+        currency.approve(dmAddr, smallAmount);
+
+        uint128 startTime = uint128(block.timestamp);
+        (uint256 cloneId, uint256 protoId) = dm.duplicate(nftAddr, nftId, currencyAddr, smallAmount, false, 0);
+        uint128 worth = getCloneShape(cloneId).worth;
+
+        vm.stopPrank();
+
+        vm.warp(block.timestamp + time);
+        vm.roll(block.number + (time/12));
```

Comment:
probably best to remove `vm.roll()` as the code doesn't depend on `block.number`. Otherwise all looks good, feel free to merge.

Reply:
fair, got it removed, i initially put it in just to see if there was any unforeseen change in behaviour.